br/tests/br_pitr_failpoint/run.sh

```diff
@@ -0,0 +1,129 @@
+#!/bin/bash
+#
+# Copyright 2024 PingCAP, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu
+. run_services
+CUR=$(cd `dirname $0`; pwd)
+
+# const value
+PREFIX="pitr_backup_failpoint" # NOTICE: don't start with 'br' because `restart services` would remove file/directory br*.
+res_file="$TEST_DIR/sql_res.$TEST_NAME.txt"
+hint_sig_file_public=$TEST_DIR/hint_sig_file_public
+hint_sig_file_history=$TEST_DIR/hint_sig_file_history
+
+# start a new cluster
+echo "restart a services"
+restart_services
+
+# prepare the data
+echo "prepare the data"
+run_sql_file $CUR/prepare_data/ingest_repair.sql
+
+# inject some failpoints
+export GO_FAILPOINTS="github.com/pingcap/tidb/pkg/ddl/create-index-stuck-before-public=1*return(\"$hint_sig_file_public\");\
+github.com/pingcap/tidb/pkg/ddl/create-index-stuck-before-ddlhistory=1*return(\"$hint_sig_file_history\")"
+
+# prepare the intersect data
+run_sql_file $CUR/intersect_data/ingest_repair1.sql &
+sql_pid=$!
+
+# start the log backup task
+echo "start log task"
+run_br --pd $PD_ADDR log start --task-name integration_test -s "local://$TEST_DIR/$PREFIX/log"
```

Comment:
br should wait until DDL really started 🤔

Reply:
In the shell function `start_tidb`, it checks TiDB by accessing `https://$TIDB_IP:10080/status`.

Reply:
Sorry I mean wait until ADD INDEX is running

Reply:
we just need the log data from [full-backup-ts, +inf).

Reply:
Get